### PR TITLE
mobile disability rating EVSS to LH

### PR DESCRIPTION
**Pull request by aherzberg**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary
Update endpoint to retrieve Rated Disability info from VA Profile/Lighthouse source

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/4765

## Testing done
specs

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/spec/request/legacy_disability_rating_request_spec.rb
Comment (aherzberg) [inline modules/mobile/spec/request/legacy_disability_rating_request_spec.rb:1]: Had a lot of error VCRs to work with so in the effort to make our error documentation more... documented, created a spec for each one.

**Comment by aherzberg** [comment]
error specs and doc changes will come in next PR.

**Comment by aherzberg** [comment]
I guess there's not much of a reduced LOC since docs didn't count towards the number and the original disability spec had some error specs as well. I can't take out much more though without breaking something so I think I'll just need an exception for this.

File: modules/mobile/app/services/mobile/v0/disability_rating/proxy.rb [hunk 1]
```diff
@@ -6,52 +6,29 @@ module Mobile
   module V0
     module DisabilityRating
       class Proxy
-        def initialize(user)
-          @user = user
+        def initialize(icn)
```
Comment (Tonksthebear) [inline modules/mobile/app/services/mobile/v0/disability_rating/proxy.rb:9]: What was the reasoning behind no longer passing the full user object through?

**Review by Tonksthebear — DISMISSED** [review]
As you said, the vast majority of changes are all cassettes. I like renaming EVSS stuff to legacy so that specs are cleaner and easier to follow. It's very nice seeing how simple the proxy becomes with lighthouse. I don't see anything concerning so LGTM

File: modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb [hunk 2]
```diff
@@ -3,16 +3,35 @@
...
   module V0
     class DisabilityRatingController < ApplicationController
-      before_action { authorize :evss, :access? }
+      before_action do
+        if Flipper.enabled?(:mobile_lighthouse_disability_ratings, @current_user)
+          authorize :lighthouse, :access?
+        else
+          authorize :evss, :access?
+        end
+      end
```
Comment (kpethtel) [inline modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb:12]: This is just a nitpicky style preference, but can we add a blank line before the index method?

**Review by kpethtel — COMMENTED** [review]
The one thing I don't understand is that you seem to have copied over a bunch of error case cassettes that aren't being used. Is the plan to write tests for those while writing the docs?

File: modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb (hunk 2 above)
Comment (aherzberg) [reply modules/mobile/app/controllers/mobile/v0/disability_rating_controller.rb:12]: no problem. rubymine likes to add newlines that aren't real lines to show annotations like who wrote it which is always confusing.

File: modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          end
+        end
+
+        def parse_date(date)
+          return nil unless date
+
+          DateTime.strptime(date, '%Y-%m-%d').iso8601
+        end
+
+        def parse_rating(rating)
```
[the remark this answers is not in the document]
Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb:37]: fixed

File: modules/mobile/app/services/mobile/v0/disability_rating/proxy.rb (hunk 1 above)
Comment (aherzberg) [reply modules/mobile/app/services/mobile/v0/disability_rating/proxy.rb:9]: the evss services the old one used required the whole user object while the new one just needs the icn.

File: modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        end
+
+        private
+
+        def individual_ratings(response)
+          response['individual_ratings'].map do |rating|
+            Mobile::V0::IndividualRating.new(
+              decision: rating['decision'],
+              effective_date: parse_date(rating['effective_date']),
+              rating_percentage: rating['rating_percentage']&.to_i(),
```
Comment (kpethtel) [inline modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb:25]: Sorry to continue being super nitpicky, but I think your IDE may have autocompleted the parens here.

**Comment by aherzberg** [comment]
@kpethtel I originally had specs to use all those VCR cassettes and test error states but removed them to try and reduce LOC. I'll bring them back in the next PR.

File: modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb (hunk 4 above)
Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/adapters/disability_rating.rb]: all good. fixed

**Review by kpethtel — APPROVED** [review]
🚀

**Review by ryan-mcneil — APPROVED** [review]
LGTM. Overriding LOC check and merging 🚀

**Pull request merged**